dev/revapi

```diff
@@ -0,0 +1,88 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+export REVAPI_VERSION=0.12.0
+export OLD_VERSION=1.5.2
+
+# Go to the root directory
+ROOT_DIR="$(cd "`dirname "$0"`"/..; pwd)"
+cd "$ROOT_DIR"
+
+REVAPI_WORKING_DIR="$ROOT_DIR/lib/temp/revapi"
+
+mkdir -p $REVAPI_WORKING_DIR
+
+export REVAPI_ZIP="$ROOT_DIR"/lib/revapi-standalone-${REVAPI_VERSION}-standalone.zip
+
+if test -x "$JAVA_HOME/bin/java"; then
+  declare java_cmd="$JAVA_HOME/bin/java"
+else
+  declare java_cmd=java
+fi
+
+REVAPI_URL=https://repo1.maven.org/maven2/org/revapi/revapi-standalone/${REVAPI_VERSION}/revapi-standalone-${REVAPI_VERSION}-standalone.zip
+
+if [ ! -f "$REVAPI_ZIP" ]; then
+  printf "Downloading revapi\n"
+  REVAPI_ZIP_DL="${REVAPI_ZIP}.part"
+  if [ $(command -v curl) ]; then
+    curl -L --silent "${REVAPI_URL}" > "$REVAPI_ZIP_DL" && mv "$REVAPI_ZIP_DL" "$REVAPI_ZIP"
+  elif [ $(command -v wget) ]; then
+    wget --quiet ${REVAPI_URL} -O "$REVAPI_ZIP_DL" && mv "$REVAPI_ZIP_DL" "$REVAPI_ZIP"
+  else
+    printf "You do not have curl or wget installed, please install revapi manually.\n"
+    exit 1
+  fi
+fi
+
+unzip -q -o "$REVAPI_ZIP" -d "$ROOT_DIR/lib"
+
+export REVAPI_HOME="$ROOT_DIR/lib/revapi-$REVAPI_VERSION"
+
+REVAPI_PROJECTS="api core parquet orc common data"
+
+REVAPI_OUT=lib/revapi.report
+rm "$REVAPI_OUT"
+rm -f "$REVAPI_OUT"
+for REVAPI_PROJECT in $REVAPI_PROJECTS; do
+  echo "========================"
+  echo "Checking iceberg-$REVAPI_PROJECT"
+  echo "Downloading iceberg-$REVAPI_PROJECT $OLD_VERSION"
+  if [ $(command -v curl) ]; then
+    curl -L --silent "https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-$REVAPI_PROJECT/$OLD_VERSION/iceberg-$REVAPI_PROJECT-$OLD_VERSION.jar" > "$REVAPI_WORKING_DIR/iceberg-$REVAPI_PROJECT-$OLD_VERSION.jar"
+  elif [ $(command -v wget) ]; then
+    wget --quiet "https://repo1.maven.org/maven2/org/apache/iceberg/iceberg-$REVAPI_PROJECT/$OLD_VERSION/iceberg-$REVAPI_PROJECT-$OLD_VERSION.jar" -O "$REVAPI_REVAPI_WORKING_DIR/iceberg-$REVAPI_PROJECT-$OLD_VERSION.jar"
+  else
+    printf "You do not have curl or wget installed, please download artifact manually.\n"
+    exit 1
+  fi
+  $REVAPI_HOME/revapi.sh --extensions=org.revapi:revapi-java:0.28.1,org.revapi:revapi-reporter-text:0.15.0 --old "$REVAPI_WORKING_DIR/iceberg-$REVAPI_PROJECT-$OLD_VERSION.jar" --new $REVAPI_PROJECT/build/libs/iceberg-$REVAPI_PROJECT*SNAPSHOT.jar -c dev/.revapi.json -Drevapi.java.missing-classes.behavior=ignore -Drevapi.reporter.text.minCriticality=error -Drevapi.reporter.text.append=true -Drevapi.reporter.text.output=$REVAPI_OUT -Drevapi.reporter.text.minSeverity=BREAKING -Drevapi.reporter.text.template=dev/.revapi.template
```

Comment:
Should we add a check to ensure that the snapshot files are there:

```
 ./dev/revapi         
rm: lib/revapi.report: No such file or directory
========================
Checking iceberg-api
Downloading iceberg-api 1.5.2
09:56:05.771 INFO  Downloading checked archives
Exception in thread "main" java.lang.IllegalArgumentException: New API files '/Users/fokkodriesprong/Desktop/iceberg/api/build/libs/iceberg-api*SNAPSHOT.jar' does not exist.
	at org.revapi.standalone.Main.checkCanRead(Main.java:508)
	at org.revapi.standalone.Main.convertPaths(Main.java:422)
	at org.revapi.standalone.Main.main(Main.java:272)
grep: lib/revapi.report: No such file or directory
========================
Checking iceberg-core
Downloading iceberg-core 1.5.2
09:56:06.360 INFO  Downloading checked archives
Exception in thread "main" java.lang.IllegalArgumentException: New API files '/Users/fokkodriesprong/Desktop/iceberg/core/build/libs/iceberg-core*SNAPSHOT.jar' does not exist.
	at org.revapi.standalone.Main.checkCanRead(Main.java:508)
	at org.revapi.standalone.Main.convertPaths(Main.java:422)
	at org.revapi.standalone.Main.main(Main.java:272)
grep: lib/revapi.report: No such file or directory
========================
Checking iceberg-parquet
Downloading iceberg-parquet 1.5.2
09:56:06.749 INFO  Downloading checked archives
Exception in thread "main" java.lang.IllegalArgumentException: New API files '/Users/fokkodriesprong/Desktop/iceberg/parquet/build/libs/iceberg-parquet*SNAPSHOT.jar' does not exist.
	at org.revapi.standalone.Main.checkCanRead(Main.java:508)
	at org.revapi.standalone.Main.convertPaths(Main.java:422)
	at org.revapi.standalone.Main.main(Main.java:272)
grep: lib/revapi.report: No such file or directory
========================
Checking iceberg-orc
Downloading iceberg-orc 1.5.2
09:56:07.145 INFO  Downloading checked archives
Exception in thread "main" java.lang.IllegalArgumentException: New API files '/Users/fokkodriesprong/Desktop/iceberg/orc/build/libs/iceberg-orc*SNAPSHOT.jar' does not exist.
	at org.revapi.standalone.Main.checkCanRead(Main.java:508)
	at org.revapi.standalone.Main.convertPaths(Main.java:422)
	at org.revapi.standalone.Main.main(Main.java:272)
grep: lib/revapi.report: No such file or directory
========================
Checking iceberg-common
Downloading iceberg-common 1.5.2
09:56:07.499 INFO  Downloading checked archives
Exception in thread "main" java.lang.IllegalArgumentException: New API files '/Users/fokkodriesprong/Desktop/iceberg/common/build/libs/iceberg-common*SNAPSHOT.jar' does not exist.
	at org.revapi.standalone.Main.checkCanRead(Main.java:508)
	at org.revapi.standalone.Main.convertPaths(Main.java:422)
	at org.revapi.standalone.Main.main(Main.java:272)
grep: lib/revapi.report: No such file or directory
========================
Checking iceberg-data
Downloading iceberg-data 1.5.2
09:56:07.861 INFO  Downloading checked archives
Exception in thread "main" java.lang.IllegalArgumentException: New API files '/Users/fokkodriesprong/Desktop/iceberg/data/build/libs/iceberg-data*SNAPSHOT.jar' does not exist.
	at org.revapi.standalone.Main.checkCanRead(Main.java:508)
	at org.revapi.standalone.Main.convertPaths(Main.java:422)
	at org.revapi.standalone.Main.main(Main.java:272)
grep: lib/revapi.report: No such file or directory
revapi check passed
```

Reply:
Good point. Actually we should check both old and new artifact files. In your case, for instance, the problem is probably because you didn't build (at least with `gradlew build -x test -x integrationTest`) the artifacts before executing `dev/revapi`: the script expects the SNAPSHOT artifacts present.

Let me add a test to be sure about that.

Reply:
For some reason, I'm unable to trigger the Rev-API: https://github.com/apache/iceberg/pull/10408. Also locally, I'm unable to reproduce it.

Let me do some investigation

Reply:
I talked with @nastra . In the mean time, and in order to move forward fast and clear, I'm working on a gradle revapi plugin fix. So, let's keep this PR as "fallback" but we can try to fix and release gradle revapi plugin (@nastra knows how to push a gradle plugin release).

Reply:
Sounds good to me! Thanks for the update 👍

Reply:
I updated the script to check old/new artifacts. Can you please do a new run ?

Reply:
@Fokko in the mean time, I'm checking #10408